File: code/scala/src/main/scala/group/pals/desktop/app/apksigner/utils/Sys.scala
/*
 *    Copyright (C) 2012 Hai Bison
 *
 *    See the file LICENSE at the root directory of this project for copying
 *    permission.
 */

package group.pals.desktop.app.apksigner.utils

import java.io.File

/**
 * System utilities.
 *
 * @author Hai Bison
 * @since v1.9 beta
 */
object Sys {

    /**
     * Debug flag.
     */
    lazy final val DEBUG = false

    /**
     * The app name.
     */
    lazy final val APP_NAME = "apk-signer"

    /**
     * The app version code.
     */
    lazy final val APP_VERSION_CODE = 43

    /**
     * The app version name.
     */
    lazy final val APP_VERSION_NAME = "1.9 beta"

    /**
     * Gets app .jar file.
     */
    def appJar(): File = {
        new File(Sys.getClass().getProtectionDomain().getCodeSource()
            .getLocation().toString())
    } // appJar()

    /**
     * Gets app root directory.
     */
    def appDir(): File = appJar().getParentFile()

}// Sys

File: code/jruby/apk-signer/utils/sys.rb
##
# Copyright (C) 2012 Hai Bison
#
# See the file LICENSE at the root directory of this project for copying
# permission.
#

##
# System utilities.
#
# * **Author:** Hai Bison
# * **Since:** v2.0 beta
#
class Sys

    ##
    # Debug flag.
    #
    DEBUG = false

    ##
    # The app name.
    #
    APP_NAME = "apk-signer"

    ##
    # The app version code.
    #
    APP_VERSION_CODE = 46

    ##
    # The app version name.
    #
    APP_VERSION_NAME = "2.0 beta"

    ##
    # The app directory.
    #
    APP_DIR = File.dirname File.dirname $0

end # Sys
